extensions/omni_txn/omni_txn.c

```diff
@@ -27,16 +68,23 @@ Datum retry(PG_FUNCTION_ARGS) {
   if (!PG_ARGISNULL(1)) {
     max_attempts = PG_GETARG_INT32(1);
   }
+  bool gather_backoff_values = true;
+  if (!PG_ARGISNULL(2)) {
+    gather_backoff_values = PG_GETARG_BOOL(2);
+  }
 
+  // make sure that we are not collecting backoff values from another
+  // transaction. Frees up existing memory in order not to leak it from
+  // existing calls.
+  list_free(backoff_values);
```

Comment:
Should we only free and reset if we're gathering and leave it intact if we aren't?

Reply:
good point